tensorflow/core/lib/io/buffered_inputstream.cc

```diff
@@ -60,25 +60,38 @@ Status BufferedInputStream::ReadLineHelper(StringType* result,
                                            bool include_eol) {
   result->clear();
   Status s;
+  size_t start_pos = pos_;
   while (true) {
     if (pos_ == limit_) {
+      if (start_pos != pos_) {
```

Comment:
This gets repeated a few times and makes the logic pretty hard to follow.

Is it enough to call result->reserve when we fill the buffer? Probably OK if a bit of extra memory is reserved for the string.